gnomad_constraint/pipeline/constraint_pipeline.py

```diff
@@ -0,0 +1,158 @@
+    DATA_TYPES,
+    VERSIONS,
+    annotated_context_ht,
+    get_logging_path,
+    get_preprocessed_ht,
+    get_sites_resource,
+)
+from gnomad_constraint.utils.constraint import (
+    add_vep_context_annotations,
+    prepare_ht_for_constraint_calculations,
+)
+
+logging.basicConfig(
+    format="%(asctime)s (%(name)s %(lineno)s): %(message)s",
+    datefmt="%m/%d/%Y %I:%M:%S %p",
+)
+logger = logging.getLogger("constraint_pipeline")
+logger.setLevel(logging.INFO)
+
+
+def main(args):
+    """Execute the constraint pipeline."""
+    hl.init(log="/constraint_pipeline.log")
+
+    test = args.test
+    overwrite = args.overwrite
+    # TODO: gnomAD v4 is still in production, for now this will only use 2.1.1.
+    version = args.version
+    if version not in VERSIONS:
+        version = CURRENT_VERSION
+        logger.warning(
+            "The requested version of resource Tables are not exist, will use gnomAD"
+            " v2.1.1 as default."
+        )
+
+    try:
+        if args.preprocess_data:
+            logger.info("Adding VEP context annotations...")
+            # TODO: Need to add function that annotates methylation, coverage, and gerp in the vep context table.
+            context_ht = annotated_context_ht.versions[version].ht()
+            # Add annotations used in constraint calculations.
+            for data_type in DATA_TYPES:
+                if data_type != "context":
+                    ht = get_sites_resource(data_type, version).ht()
+                else:
+                    ht = context_ht
+
+                # Filtering the Table to chr20, chrX, and chrY for testing if applicable
+                if test:
+                    rg = get_reference_genome(context_ht.locus)
+                    contigs_keep = [
+                        hl.parse_locus_interval(c, reference_genome=rg)
+                        for c in [rg.contigs[19], rg.x_contigs[0], rg.y_contigs[0]]
+                    ]
+                    logger.info(
+                        "Filtering the %s HT to chr20, chrX, and chrY for testing...",
+                        data_type,
+                    )
+                    ht = hl.filter_intervals(ht, contigs_keep)
+
+                # Add annotations from VEP context Table to genome and exome Tables.
+                if data_type != "context":
+                    ht = add_vep_context_annotations(ht, context_ht)
+
+                # Filter input Table and add annotations used in constraint calculations.
+                ht = prepare_ht_for_constraint_calculations(ht)
+                # Filter to locus that is on an autosome or in a pseudoautosomal region.
+                ht.filter(ht.locus.in_autosome_or_par()).write(
+                    get_preprocessed_ht(data_type, version, "autosome_par", test).path,
+                    overwrite=overwrite,
+                )
+
+                # Sex chromosomes are analyzed separately, since they are biologically different from the autosomes.
```

Comment:
```suggestion
                # Sex chromosomes are analyzed separately, since they are biologically
                # different from the autosomes.```